spec/unit/analytics/ga4_import/elastic_search_relevancy_serialiser_spec.rb

```diff
@@ -0,0 +1,26 @@
+require "spec_helper"
+
+RSpec.describe Analytics::Ga4Import::ElasticSearchRelevancySerialiser do
+  describe "#relevance" do
+    let(:consolidated_data) { { "/example" => 40, "/example2" => 20, "/example3" => 100, "/example3/child" => 100 } }
+    let(:relevancy_calculator) { described_class.new(consolidated_data) }
+    let(:relevancy_data) { relevancy_calculator.relevance.map { |rel| JSON.parse(rel) } }
+
+    it "returns an array of JSON object pairs of index and page data" do
+      relevancy_data.each_slice(2) do |index_data, page_data|
+        expect(index_data).to include("index")
+        expect(page_data).to include("path_components", "rank_14", "vc_14", "vf_14")
+      end
+    end
+
+    it "specifies the page rank, views and proportion of overall views for a page" do
+      example_page_data = relevancy_data.each_slice(2).map { |_, page_data| page_data }.find { |data| data["path_components"].include?("/example") }
+      expect(example_page_data).to include("rank_14" => 1, "vc_14" => 40, "vf_14" => 0.15384615384615385)
```

Comment:
This test would could be a bit easier to follow with tailored data going into it and avoiding the each_slice

```
consolidated_data = [["/example", 30], ["/other", 10]]
relevancy_data = described_class.new(consolidated_data).relevance
example_page_data = JSON.parse(relevancy_data.first)
expect(example_page_data).to include("rank_14" => 1, "vc_14" => 30, "vf_14" => 0.75)
```